cirq/qis/measures.py

```diff
@@ -38,36 +56,211 @@ def fidelity(state1: np.ndarray, state2: np.ndarray) -> float:
     Args:
         state1: The first state.
         state2: The second state.
+        qid_shape: The qid shape of the given states.
+        validate: Whether to check if the given states are valid quantum states.
+        atol: Absolute numerical tolerance to use for validation.
+
+    Returns:
+        The fidelity.
+
+    Raises:
+        ValueError: The qid shape of the given states was not specified and
+            could not be inferred.
+        ValueError: Invalid quantum state.
     """
-    if len(state1.shape) == 1 and len(state2.shape) == 1:
+    # Two ints
+    if isinstance(state1, int) and isinstance(state2, int):
+        if validate and qid_shape is not None:
+            dim = np.prod(qid_shape)
+            if state1 >= dim:
```

Comment:
Do we want to guard against negative states too?

Reply:
Done.